x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -592,6 +599,29 @@ private static Ordinal tailOrdinal(List<SearchHit> hits, Criterion<BoxedQueryReq
         return criterion.ordinal(hits.get(hits.size() - 1));
     }
 
+    // Wrapper method of matcher.match() which is called for every sub query in the sequence query
+    // and for each subquery every "fetch_size" docs. Doing RAM accounting on object creation is
+    // expensive, so we just calculate the difference in bytes of the total memory that the matcher's
+    // structure occupy, before and after the match() call.
+    private boolean match(int stage, Iterable<Tuple<KeyAndOrdinal, HitReference>> hits) {
```

Comment:
Since all the data and thus tracking is done in the sequence matcher, it's easier to move the accounting there.
Keep the `TumblingWindow` intact, move the tracking directly into `matcher.match` (no need to wrap it) including clearing the memory.